lib/oneview-sdk/resource/api500/c7000/volume.rb

```diff
@@ -62,16 +62,19 @@ def delete(flag = :all)
         # @param [OneviewSDK::StoragePool] storage_pool Storage pool.
         def set_storage_pool(storage_pool)
           assure_uri(storage_pool)
-          @data['properties'] ||= {}
-          @data['properties']['storagePool'] = storage_pool['uri']
+          @data['properties']['storagePool'] = storage_pool['uri'] if @data['properties']
```

Comment:
As far as it fixes the PUT it may break the POST.

Reply:
I'm concerned if it won't cause any breakage in already existing code

Reply:
I understand. Maybe another way would be inside the update method, delete the properties attribute. What do you think?

Reply:
Why not keeping the code how have been working before?
Because this helper method is just to be used before of the create (POST) action. For update, the user cannot set the storage_pool.
Maybe documenting that be sufficient.